packages/cli/templates/vue3/src/components/FieldPlugin.vue

```diff
@@ -0,0 +1,76 @@
+<script setup lang="ts">
+import ModalToggle from './ModalToggle.vue'
+import Counter from './Counter.vue'
+import AssetSelector from './AssetSelector.vue'
+
+import { PluginActions, PluginState } from '@storyblok/field-plugin'
+
+const props = defineProps<{
+  actions: PluginActions
+  data: PluginState
+}>()
+
+function closeModal() {
+  props.actions.setModalOpen(false)
+}
+</script>
+
+<template>
+  <div class="container">
+    <button
+      v-if="props.data.isModalOpen"
+      type="button"
+      class="btn btn-close"
```

Comment:
Yes, it might be better to not split it. It is a matter of copy-paste if the users desire to separate this. I would wait for the feedback if it's crucial to split it, or if we see that it is very common we can reconsider.